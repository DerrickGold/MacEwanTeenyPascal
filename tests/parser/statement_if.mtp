begin
	(*if statement with simple condition*)
	if (10) then
		sample := 0;

	(*if statement with multiply op expression*)
	if (10 * 20) then
		sample := 1;

	(*if statement with binary op*)
	if (10 * 20 + 10 div 30) then
		sample := 2;

	(*if statement with binary op*)
	if (10 * 20 + 10 div 30 > 20 mod 2) then
		sample := 3;


	(*if statement with multiple statements in block*)
	if (10) then begin
		sample2 := 0;
		sample := 4
	end;

	(*if statement with else statement*)
	if (10) then
		sample := 5
	else
		sample := 6;

	(*if else using blocks*)
	if (10) then begin
		sample := 7;
		sample2 := 1;
	end
	else begin
		sample := 8;
		sample2 := 2;
	end;

	(*if without parenthesis*)
	if 10 * 20 = 220 - 20 then
		sample := 9


end
.