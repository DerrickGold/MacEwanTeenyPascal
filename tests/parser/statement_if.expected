[Head] (Block)
    [Child] (Statement List)
        [Child] (If Statement)
            [Child] 3, TOK_NUM, 10, (Condition, Expression, Simple Expression, Factor, Constant)
            [Child] (Assign Statement)
                [Child] 4, TOK_ID, sample, (Simple Name)
                [Child] 4, TOK_NUM, 0, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 7, TOK_STAR, *, (Condition, Expression, Simple Expression, Factor, Multiply Operator)
                [Child] 7, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 7, TOK_NUM, 20, (Factor, Constant, Term)
            [Child] (Assign Statement)
                [Child] 8, TOK_ID, sample, (Simple Name)
                [Child] 8, TOK_NUM, 1, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 11, TOK_PLUS, +, (Condition, Expression, Simple Expression, Factor, Binary Adding Operator)
                [Child] 11, TOK_STAR, *, (Simple Expression, Multiply Operator)
                    [Child] 11, TOK_NUM, 10, (Factor, Constant, Term)
                    [Child] 11, TOK_NUM, 20, (Factor, Constant, Term)
                [Child] 11, TOK_KEY_DIV, div, (Multiply Operator)
                    [Child] 11, TOK_NUM, 10, (Factor, Constant, Term)
                    [Child] 11, TOK_NUM, 30, (Factor, Constant, Term)
            [Child] (Assign Statement)
                [Child] 12, TOK_ID, sample, (Simple Name)
                [Child] 12, TOK_NUM, 2, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 15, TOK_GREATER, >, (Condition, Expression, Simple Expression, Factor, Relational Operator)
                [Child] 15, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 15, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 15, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 15, TOK_NUM, 20, (Factor, Constant, Term)
                    [Child] 15, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 15, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 15, TOK_NUM, 30, (Factor, Constant, Term)
                [Child] 15, TOK_KEY_MOD, mod, (Simple Expression, Multiply Operator)
                    [Child] 15, TOK_NUM, 20, (Factor, Constant, Term)
                    [Child] 15, TOK_NUM, 2, (Factor, Constant, Term)
            [Child] (Assign Statement)
                [Child] 16, TOK_ID, sample, (Simple Name)
                [Child] 16, TOK_NUM, 3, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 20, TOK_NUM, 10, (Condition, Expression, Simple Expression, Factor, Constant)
            [Child] (Block, Block Statement)
                [Child] (Statement List)
                    [Child] (Assign Statement)
                        [Child] 21, TOK_ID, sample2, (Simple Name)
                        [Child] 21, TOK_NUM, 0, (Expression, Simple Expression, Factor, Constant)
                    [Sibling] (Assign Statement)
                        [Child] 22, TOK_ID, sample, (Simple Name)
                        [Child] 22, TOK_NUM, 4, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 26, TOK_NUM, 10, (Condition, Expression, Simple Expression, Factor, Constant)
            [Child] (Assign Statement)
                [Child] 27, TOK_ID, sample, (Simple Name)
                [Child] 27, TOK_NUM, 5, (Expression, Simple Expression, Factor, Constant)
            [Child] (Assign Statement)
                [Child] 29, TOK_ID, sample, (Simple Name)
                [Child] 29, TOK_NUM, 6, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 32, TOK_NUM, 10, (Condition, Expression, Simple Expression, Factor, Constant)
            [Child] (Block, Block Statement)
                [Child] (Statement List)
                    [Child] (Assign Statement)
                        [Child] 33, TOK_ID, sample, (Simple Name)
                        [Child] 33, TOK_NUM, 7, (Expression, Simple Expression, Factor, Constant)
                    [Sibling] (Assign Statement)
                        [Child] 34, TOK_ID, sample2, (Simple Name)
                        [Child] 34, TOK_NUM, 1, (Expression, Simple Expression, Factor, Constant)
                    [Sibling] (Null Statement)
            [Child] (Block, Block Statement)
                [Child] (Statement List)
                    [Child] (Assign Statement)
                        [Child] 37, TOK_ID, sample, (Simple Name)
                        [Child] 37, TOK_NUM, 8, (Expression, Simple Expression, Factor, Constant)
                    [Sibling] (Assign Statement)
                        [Child] 38, TOK_ID, sample2, (Simple Name)
                        [Child] 38, TOK_NUM, 2, (Expression, Simple Expression, Factor, Constant)
                    [Sibling] (Null Statement)
        [Sibling] (If Statement)
            [Child] 42, TOK_EQ, =, (Condition, Expression, Relational Operator)
                [Child] 42, TOK_STAR, *, (Simple Expression, Multiply Operator)
                    [Child] 42, TOK_NUM, 10, (Factor, Constant, Term)
                    [Child] 42, TOK_NUM, 20, (Factor, Constant, Term)
                [Child] 42, TOK_MINUS, -, (Simple Expression, Binary Adding Operator)
                    [Child] 42, TOK_NUM, 220, (Simple Expression, Factor, Constant)
                    [Child] 42, TOK_NUM, 20, (Factor, Constant)
            [Child] (Assign Statement)
                [Child] 43, TOK_ID, sample, (Simple Name)
                [Child] 43, TOK_NUM, 9, (Expression, Simple Expression, Factor, Constant)
